server/src/main/java/org/opensearch/action/admin/indices/shrink/TransportResizeAction.java

```diff
@@ -138,26 +146,93 @@ protected void clusterManagerOperation(
         // there is no need to fetch docs stats for split but we keep it simple and do it anyway for simplicity of the code
         final String sourceIndex = indexNameExpressionResolver.resolveDateMathExpression(resizeRequest.getSourceIndex());
         final String targetIndex = indexNameExpressionResolver.resolveDateMathExpression(resizeRequest.getTargetIndexRequest().index());
-        client.admin()
-            .indices()
-            .prepareStats(sourceIndex)
-            .clear()
-            .setDocs(true)
-            .setStore(true)
-            .execute(ActionListener.delegateFailure(listener, (delegatedListener, indicesStatsResponse) -> {
-                CreateIndexClusterStateUpdateRequest updateRequest = prepareCreateIndexRequest(resizeRequest, state, i -> {
-                    IndexShardStats shard = indicesStatsResponse.getIndex(sourceIndex).getIndexShards().get(i);
-                    return shard == null ? null : shard.getPrimary().getDocs();
-                }, indicesStatsResponse.getPrimaries().store, sourceIndex, targetIndex);
-                createIndexService.createIndex(
-                    updateRequest,
-                    ActionListener.map(
-                        delegatedListener,
-                        response -> new ResizeResponse(response.isAcknowledged(), response.isShardsAcknowledged(), updateRequest.index())
-                    )
-                );
-            }));
 
+        IndexMetadata indexMetadata = state.metadata().index(sourceIndex);
+        if (state.metadata().isSegmentReplicationEnabled(sourceIndex)
+            && Integer.valueOf(indexMetadata.getSettings().get("index.number_of_replicas")) > 0
+            && resizeRequest.getResizeType().equals(ResizeType.SHRINK)) {
+            client.admin()
+                .indices()
+                .prepareStats(sourceIndex)
+                .setRefresh(true)
```

Comment:
This is setting the flag for returning refresh stats, why do we need those?  If you do need that, you are clearing flags immediately after so this wont' fetch them.